Assignment/test.c

```diff
@@ -0,0 +1,126 @@
+// ANSI color code
+#define GREEN_TEXT "\033[0;32m"
+#define RED_TEXT "\033[0;31m"
+#define RESET_TEXT "\033[0m"
+
+int IsValid (const char* n);
+void DecToBinaryTest ();
+void DecToHexaTest ();
+void UserInput ();
+
+int IsValid (const char* n) {
+   if (n == NULL || *n == '\0') return 0;
+   char* endptr;
+   long num = strtol (n, &endptr, 10);
+   if (*endptr != '\0' || num < INT_MIN || num > INT_MAX) return 0;
+   return 1;
+}
+
+void DecToBinaryTest () {
+   const char* input[] = { "0", "5", "255", "-255", "2147483647", "-2147483647", "*", "abc56" };
+   const char* expectedOutput[] = { "00000000", "00000101", "0000000011111111", "1111111100000001",
+                                    "01111111111111111111111111111111", "10000000000000000000000000000001",
+                                    INVALID_INPUT, INVALID_INPUT };
+   int testCount = sizeof (input) / sizeof (input[0]);
+   char actualBinary[BINARY_BUFFER_SIZE];
+   printf ("---Decimal to Binary---\n\n");
+   printf ("| %-12s | %-35s | %-40s | %-19s |\n", "Input", "Expected Output", "Actual Output", "Test Case Result");
+   printf ("|--------------|-------------------------------------|------------------------------------------|---------------------|\n");
+   for (int i = 0; i < testCount; i++) {
+      int value = atoi (input[i]);
+      char* actualBinaryOutput = IsValid (input[i]) ? DecToBinary (value, actualBinary) : INVALID_INPUT;
+      const char* result = (strcmp (actualBinaryOutput, expectedOutput[i]) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT;
+      printf ("| %-12s | %-35s | %-40s | %-30s |\n", input[i], expectedOutput[i], actualBinaryOutput, result);
+   }
+   printf ("\n\n");
+}
+
+void DecToHexaTest () {
+   const char* input2[] = { "0", "5", "255", "-255", "2147483647", "-2147483647", "*", "vini" };
+   const char* expectedOutput[] = { "00", "05", "00FF", "FF01", "7FFFFFFF", "80000001", INVALID_INPUT, INVALID_INPUT };
+   int testCount = sizeof (input2) / sizeof (input2[0]);
+   char actualHexa[HEXA_BUFFER_SIZE];
+   printf ("---Decimal to Hexadecimal---\n\n");
+   printf ("| %-12s | %-20s | %-20s | %-20s |\n", "Input", "Expected Output", "Actual Output", "Test Case Result");
+   printf ("|--------------|----------------------|----------------------|----------------------|\n");
+   for (int i = 0; i < testCount; i++) {
+      int value = atoi (input2[i]);
+      char* actualHexaOutput = IsValid (input2[i]) ? DecToHexa (value, actualHexa) : INVALID_INPUT;
+      const char* result = (strcmp (actualHexaOutput, expectedOutput[i]) == 0) ? GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "Fail" RESET_TEXT;
+      printf ("| %-12s | %-20s | %-20s | %-31s |\n", input2[i], expectedOutput[i], actualHexaOutput, result);
+   }
+}
+
+void UserInput () {
+   char buffer[256]; // Buffer to hold user input
+   int num, validInput = 0;
+   while (!validInput) {
+      printf ("\nEnter a decimal number: ");
+      if (fgets (buffer, sizeof (buffer), stdin) != NULL) {

```

Comment:
```suggestion
      if (fgets (buffer, sizeof (buffer), stdin) == NULL)  continue;

```
See [Guard](https://en.wikipedia.org/wiki/Guard_(computer_science)).